geode-core/src/main/java/org/apache/geode/cache/execute/FunctionService.java

```diff
@@ -72,7 +72,7 @@ protected FunctionService(FunctionExecutionService functionExecutionService) {
    * @throws FunctionException if the region passed in is null
    * @since GemFire 6.0
    */
-  public static Execution onRegion(Region region) {
+  public static <I, O, A> Execution<I, O, A> onRegion(Region region) {
```

Comment:
@upthewaterspout Is this an ok change to the User API for a minor release? I'm guessing this might need to wait for Geode 2.0.

Reply:
Thanks to type erasure it is (at least theoretically) possible to introduce generics without breaking binary compatibility...